src/coreclr/jit/lower.cpp

```diff
@@ -2783,6 +2783,22 @@ GenTree* Lowering::LowerCompare(GenTree* cmp)
     }
 #endif
 
+    // Optimizes (X & 1) == 1 to (X & 1)
+    // GTF_RELOP_JMP_USED is used to make sure the optimization is used for return statements only.
```

Comment:
I would look into moving this optimization to [morph](https://github.com/dotnet/runtime/blob/e233b7b5b00d9b80f155cd484e246dfc24fdf801/src/coreclr/jit/morph.cpp#L13402). We tend not to put general peepholes like that into lowering.

Reply:
My impression that optimizations which depend on GTF_RELOP_JMP_USED are better be in lower, after various copy-props - at least in my recent similar opt when I moved it to lower I got better diffs

Reply:
Would be interesting to check what the diffs look like for this case. I cannot think off the top of my head why would we catch more cases in lower (well, modulo the "missed remorph" cases).

Relops under jumps are marked `NO_CSE`, so, in theory, we should have about the same number of `!GTF_RELOP_JMP_USED` ones before and after optimizations (modulo dead code and such).

It is a bit better to catch these things in moprh in general because we help CSE make better decisions that way, but if it turns out the diffs say we should do it in lower, so be it.